In [2]:
import pandas as pd

In [3]:
orders= pd.read_excel("c:\\Users\\Charmaine\\Downloads\\Global Superstore Data.xlsx", sheet_name= "Orders")
returns = pd.read_excel("c:\\Users\\Charmaine\\Downloads\\Global Superstore Data.xlsx", sheet_name= "Returns")
people = pd.read_excel("c:\\Users\\Charmaine\\Downloads\\Global Superstore Data.xlsx", sheet_name= "People")

In [4]:
print(orders.head())
print(returns.head())
print (people.head())

   Row ID                Order ID Order Date  Ship Date       Ship Mode  \
0   24599  IN-2017-CA120551-42816 2017-03-22 2017-03-29  Standard Class   
1   29465  ID-2015-BD116051-42248 2015-09-01 2015-09-04    Second Class   
2   24598  IN-2017-CA120551-42816 2017-03-22 2017-03-29  Standard Class   
3   24597  IN-2017-CA120551-42816 2017-03-22 2017-03-29  Standard Class   
4   29464  ID-2015-BD116051-42248 2015-09-01 2015-09-04    Second Class   

  Customer ID    Customer Name      Segment  Postal Code   City  ...  \
0   CA-120551  Cathy Armstrong  Home Office          NaN  Herat  ...   
1   BD-116051     Brian Dahlen     Consumer          NaN  Herat  ...   
2   CA-120551  Cathy Armstrong  Home Office          NaN  Herat  ...   
3   CA-120551  Cathy Armstrong  Home Office          NaN  Herat  ...   
4   BD-116051     Brian Dahlen     Consumer          NaN  Herat  ...   

    Product ID                                       Product Name  \
0  FUR-BO-4861                    Ikea Library 

In [5]:
# Checking the size of the orders, returns, and people dataframes
print(orders.shape)
print(returns.shape)
print(people.shape)

(51290, 24)
(2033, 3)
(24, 2)


In [6]:
# Checking the column names of the orders, returns, and people dataframes
print(orders.columns)
print (returns.columns)
print(people.columns)

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Postal Code', 'City',
       'State', 'Country', 'Region', 'Market', 'Product ID', 'Product Name',
       'Sub-Category', 'Category', 'Sales', 'Quantity', 'Discount', 'Profit',
       'Shipping Cost', 'Order Priority'],
      dtype='str')
Index(['Returned', 'Order ID', 'Region'], dtype='str')
Index(['Person', 'Region'], dtype='str')


In [7]:
# Checking the data types for orders column
orders.dtypes

Row ID                     int64
Order ID                     str
Order Date        datetime64[us]
Ship Date         datetime64[us]
Ship Mode                    str
Customer ID                  str
Customer Name                str
Segment                      str
Postal Code              float64
City                         str
State                        str
Country                      str
Region                       str
Market                       str
Product ID                   str
Product Name                 str
Sub-Category                 str
Category                     str
Sales                    float64
Quantity                   int64
Discount                 float64
Profit                   float64
Shipping Cost            float64
Order Priority               str
dtype: object

In [8]:
# return column data types
returns.dtypes

Returned    str
Order ID    str
Region      str
dtype: object

In [9]:
# people column data types
people.dtypes

Person    str
Region    str
dtype: object

**Getting basic statistics** 

In [10]:
orders.describe()

,Row ID,Order Date,Ship Date,Postal Code,Sales,Quantity,Discount,Profit,Shipping Cost
count,51290.00000,51290,51290,9994.000000,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000
mean,25645.50000,2016-05-11 11:52:18.436342,2016-05-15 11:09:41.306297,55190.379428,246.490581,3.476545,0.142908,28.610982,26.478567
min,1.00000,2014-01-01 00:00:00,2014-01-03 00:00:00,1040.000000,0.444000,1.000000,0.000000,-6599.978000,1.002000
25%,12823.25000,2015-06-19 00:00:00,2015-06-23 00:00:00,23223.000000,30.758625,2.000000,0.000000,0.000000,2.610000
50%,25645.50000,2016-07-08 00:00:00,2016-07-12 00:00:00,56430.500000,85.053000,3.000000,0.000000,9.240000,7.790000
75%,38467.75000,2017-05-22 00:00:00,2017-05-26 00:00:00,90008.000000,251.053200,5.000000,0.200000,36.810000,24.450000
max,51290.00000,2017-12-31 00:00:00,2018-01-07 00:00:00,99301.000000,22638.480000,14.000000,0.850000,8399.976000,933.570000
std,14806.29199,NaN,NaN,32063.693350,487.565361,2.278766,0.212280,174.340972,57.251373


In [11]:
# Returning one column ( A series)
orders["Sales"]

0        731.820
1        243.540
2        346.320
3        169.680
4        203.880
          ...   
51285      9.612
51286      4.104
51287      7.749
51288    104.364
51289      3.465
Name: Sales, Length: 51290, dtype: float64

In [12]:
# Returning multiple columns
orders[["Order ID", "Customer Name", "Sales", "Profit"]]

,Order ID,Customer Name,Sales,Profit
0,IN-2017-CA120551-42816,Cathy Armstrong,731.820,102.420
1,ID-2015-BD116051-42248,Brian Dahlen,243.540,104.490
2,IN-2017-CA120551-42816,Cathy Armstrong,346.320,13.770
3,IN-2017-CA120551-42816,Cathy Armstrong,169.680,79.680
4,ID-2015-BD116051-42248,Brian Dahlen,203.880,24.360
...,...,...,...,...
51285,ZA-2014-AS285147-41718,Alejandro Savely,9.612,-21.168
51286,ZA-2017-HG4965147-42876,Henry Goldwyn,4.104,-4.806
51287,ZA-2016-EB3870147-42499,Emily Burns,7.749,-9.051
51288,ZA-2015-JG5115147-42040,Jack Garza,104.364,-173.976


In [13]:
# Selecting rows by their positions
orders.iloc[0]

Row ID                                      24599
Order ID                   IN-2017-CA120551-42816
Order Date                    2017-03-22 00:00:00
Ship Date                     2017-03-29 00:00:00
Ship Mode                          Standard Class
Customer ID                             CA-120551
Customer Name                     Cathy Armstrong
Segment                               Home Office
Postal Code                                   NaN
City                                        Herat
State                                       Hirat
Country                               Afghanistan
Region                              Southern Asia
Market                               Asia Pacific
Product ID                            FUR-BO-4861
Product Name      Ikea Library with Doors, Mobile
Sub-Category                            Bookcases
Category                                Furniture
Sales                                      731.82
Quantity                                        2


In [14]:
# Selecting the first 5 rows
orders.iloc[0:5]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,...,Product ID,Product Name,Sub-Category,Category,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority
0,24599,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,FUR-BO-4861,"Ikea Library with Doors, Mobile",Bookcases,Furniture,731.82,2,0.0,102.42,39.66,Medium
1,29465,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,...,OFF-SU-2988,"Acme Scissors, Easy Grip",Supplies,Office Supplies,243.54,9,0.0,104.49,18.72,Medium
2,24598,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,TEC-MA-4211,"Epson Receipt Printer, White",Machines,Technology,346.32,3,0.0,13.77,14.10,Medium
3,24597,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,FUR-FU-5726,"Rubbermaid Door Stop, Erganomic",Furnishings,Furniture,169.68,4,0.0,79.68,11.01,Medium
4,29464,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,...,OFF-EN-3664,"Cameo Interoffice Envelope, with clear poly wi...",Envelopes,Office Supplies,203.88,4,0.0,24.36,5.72,Medium


**Filtering the data**


In [15]:
# Orders with a sales greater than 1000
orders[orders["Sales"]> 1000]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,...,Product ID,Product Name,Sub-Category,Category,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority
5,28879,ID-2015-AJ107801-42113,2015-04-19,2015-04-22,First Class,AJ-107801,Anthony Jacobs,Corporate,NaN,Kabul,...,FUR-TA-3420,"Bevis Conference Table, Fully Assembled",Tables,Furniture,4626.15,5,0.0,647.55,835.57,High
6,27993,IN-2017-GM144551-42948,2017-08-01,2017-08-05,Standard Class,GM-144551,Gary Mitchum,Home Office,NaN,Kabul,...,FUR-BO-3626,"Bush Classic Bookcase, Pine",Bookcases,Furniture,2070.15,5,0.0,848.70,185.01,Medium
8,29492,IN-2016-LO171701-42637,2016-09-24,2016-09-28,Standard Class,LO-171701,Lori Olson,Corporate,NaN,Kabul,...,TEC-PH-5814,"Samsung Audio Dock, VoIP",Phones,Technology,1168.44,7,0.0,443.94,92.34,Medium
10,25232,ID-2015-SS201401-42354,2015-12-16,2015-12-20,Standard Class,SS-201401,Saphhira Shifley,Corporate,NaN,Kabul,...,FUR-TA-3780,"Chromcraft Wood Table, with Bottom Storage",Tables,Furniture,1447.56,3,0.0,564.48,73.63,Medium
46,21400,IN-2016-VW217751-42632,2016-09-19,2016-09-24,Standard Class,VW-217751,Victoria Wilson,Corporate,NaN,Kandahar,...,TEC-AC-5135,"Logitech Router, Erganomic",Accessories,Technology,1470.78,6,0.0,514.62,70.16,Medium
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
51136,51240,ZA-2015-AB15146-42100,2015-04-06,2015-04-12,Standard Class,AB-15146,Aaron Bergman,Consumer,NaN,Mufulira,...,TEC-MA-5006,"Konica Inkjet, Durable",Machines,Technology,1247.40,4,0.0,74.76,81.60,Medium
51164,43519,ZA-2015-DJ3510146-42254,2015-09-07,2015-09-09,Second Class,DJ-3510146,Don Jones,Corporate,NaN,Lusaka,...,TEC-PH-5354,"Nokia Smart Phone, Cordless",Phones,Technology,1272.30,2,0.0,407.10,135.43,Medium
51165,47556,ZA-2016-TA11385146-42422,2016-02-22,2016-02-27,Standard Class,TA-11385146,Tom Ashbrook,Home Office,NaN,Lusaka,...,FUR-BO-5951,"Sauder Classic Bookcase, Traditional",Bookcases,Furniture,1743.96,4,0.0,714.96,124.45,High
51166,42448,ZA-2017-DB3615146-42993,2017-09-15,2017-09-21,Standard Class,DB-3615146,Doug Bickford,Consumer,NaN,Lusaka,...,FUR-BO-4863,"Ikea Library with Doors, Traditional",Bookcases,Furniture,2180.34,6,0.0,959.22,112.59,Medium


In [16]:
# Orders with a negative profit
orders[orders["Profit"] < 0]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,...,Product ID,Product Name,Sub-Category,Category,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority
390,6915,US-2017-BT115305-43067,2017-11-28,2017-12-03,Standard Class,BT-115305,Bradley Talbott,Home Office,NaN,Buenos Aires,...,TEC-PH-5356,"Nokia Smart Phone, with Caller ID",Phones,Technology,1277.82000,5,0.400,-0.08000,129.153,Medium
391,6755,US-2017-BW112005-42919,2017-07-03,2017-07-05,Second Class,BW-112005,Ben Wallace,Consumer,NaN,Buenos Aires,...,TEC-CO-6012,"Sharp Wireless Fax, Laser",Copiers,Technology,993.25408,7,0.402,-3.40592,109.642,High
392,6986,MX-2015-CR127305-42250,2015-09-03,2015-09-06,First Class,CR-127305,Craig Reiter,Consumer,NaN,Buenos Aires,...,TEC-CO-3604,"Brother Personal Copier, Color",Copiers,Technology,345.23736,6,0.402,-24.36264,103.941,High
393,8221,MX-2015-JH159855-42283,2015-10-06,2015-10-07,Same Day,JH-159855,Joseph Holt,Consumer,NaN,Buenos Aires,...,TEC-CO-3599,"Brother Ink, Color",Copiers,Technology,296.48840,5,0.402,-40.71160,100.933,High
396,9778,US-2017-PB191505-43091,2017-12-22,2017-12-29,Standard Class,PB-191505,Philip Brown,Consumer,NaN,Buenos Aires,...,FUR-CH-4654,"Hon Executive Leather Armchair, Adjustable",Chairs,Furniture,919.26000,5,0.400,-168.54000,77.438,Medium
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
51285,46231,ZA-2014-AS285147-41718,2014-03-20,2014-03-25,Standard Class,AS-285147,Alejandro Savely,Corporate,NaN,Harare,...,OFF-AR-5911,"Sanford Highlighters, Easy-Erase",Art,Office Supplies,9.61200,2,0.700,-21.16800,1.020,Medium
51286,50122,ZA-2017-HG4965147-42876,2017-05-21,2017-05-23,Second Class,HG-4965147,Henry Goldwyn,Corporate,NaN,Mutare,...,OFF-LA-3260,"Avery Color Coded Labels, 5000 Label Set",Labels,Office Supplies,4.10400,1,0.700,-4.80600,1.800,High
51287,50297,ZA-2016-EB3870147-42499,2016-05-09,2016-05-15,Standard Class,EB-3870147,Emily Burns,Consumer,NaN,Mutare,...,OFF-AR-5922,"Sanford Pencil Sharpener, Fluorescent",Art,Office Supplies,7.74900,1,0.700,-9.05100,1.460,Medium
51288,47164,ZA-2015-JG5115147-42040,2015-02-05,2015-02-10,Standard Class,JG-5115147,Jack Garza,Consumer,NaN,Kadoma,...,TEC-MA-5542,"Panasonic Card Printer, Durable",Machines,Technology,104.36400,2,0.700,-173.97600,7.460,Medium


In [17]:
# Finding orders from the consumer segment
orders[orders["Segment"]== "Consumer"]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,...,Product ID,Product Name,Sub-Category,Category,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority
1,29465,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,...,OFF-SU-2988,"Acme Scissors, Easy Grip",Supplies,Office Supplies,243.540,9,0.0,104.490,18.72,Medium
4,29464,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,...,OFF-EN-3664,"Cameo Interoffice Envelope, with clear poly wi...",Envelopes,Office Supplies,203.880,4,0.0,24.360,5.72,Medium
11,23222,IN-2017-AA103751-42926,2017-07-10,2017-07-15,Second Class,AA-103751,Allen Armold,Consumer,NaN,Kabul,...,OFF-AP-4747,"Hoover Toaster, Red",Appliances,Office Supplies,669.120,8,0.0,280.800,72.67,High
12,29094,IN-2015-BG110351-42275,2015-09-28,2015-10-04,Standard Class,BG-110351,Barry Gonzalez,Consumer,NaN,Kabul,...,TEC-CO-4588,"Hewlett Personal Copier, High-Speed",Copiers,Technology,427.410,3,0.0,4.230,60.91,Low
13,28265,IN-2016-AH105851-42701,2016-11-27,2016-12-01,Standard Class,AH-105851,Angele Hood,Consumer,NaN,Kabul,...,FUR-CH-4523,"Harbour Creations Chairmat, Black",Chairs,Furniture,417.420,6,0.0,45.900,58.04,High
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
51276,48014,ZA-2017-NG8430147-43063,2017-11-24,2017-11-30,Standard Class,NG-8430147,Nathan Gelder,Consumer,NaN,Harare,...,TEC-MA-5494,"Okidata Calculator, Red",Machines,Technology,14.832,1,0.7,-20.298,1.93,Medium
51279,48013,ZA-2017-NG8430147-43063,2017-11-24,2017-11-30,Standard Class,NG-8430147,Nathan Gelder,Consumer,NaN,Harare,...,OFF-AR-5923,"Sanford Pencil Sharpener, Water Color",Art,Office Supplies,7.830,1,0.7,-11.490,1.61,Medium
51284,48012,ZA-2017-NG8430147-43063,2017-11-24,2017-11-30,Standard Class,NG-8430147,Nathan Gelder,Consumer,NaN,Harare,...,OFF-BI-6397,"Wilson Jones Hole Reinforcements, Clear",Binders,Office Supplies,1.197,1,0.7,-2.373,1.06,Medium
51287,50297,ZA-2016-EB3870147-42499,2016-05-09,2016-05-15,Standard Class,EB-3870147,Emily Burns,Consumer,NaN,Mutare,...,OFF-AR-5922,"Sanford Pencil Sharpener, Fluorescent",Art,Office Supplies,7.749,1,0.7,-9.051,1.46,Medium


 **Checking missing values**

In [18]:
orders.isnull().sum()

Row ID                0
Order ID              0
Order Date            0
Ship Date             0
Ship Mode             0
Customer ID           0
Customer Name         0
Segment               0
Postal Code       41296
City                  0
State                 0
Country               0
Region                0
Market                0
Product ID            0
Product Name          0
Sub-Category          0
Category              0
Sales                 0
Quantity              0
Discount              0
Profit                0
Shipping Cost         0
Order Priority        0
dtype: int64

In [19]:
returns.isnull().sum()

Returned    0
Order ID    0
Region      0
dtype: int64

In [20]:
people.isnull().sum()

Person    0
Region    0
dtype: int64

**Checking missing data**

In [21]:
orders.duplicated().sum()

np.int64(0)

In [22]:
returns.duplicated().sum()

np.int64(63)

In [23]:
people.duplicated().sum()

np.int64(0)

In [24]:
# Removing duplicates 
returns = returns.drop_duplicates()

In [25]:
# Converting Dates from string to date
orders["Order Date"] = pd.to_datetime(orders["Order Date"])

In [26]:
orders["Ship Date"] = pd.to_datetime(orders["Ship Date"])

In [27]:
orders[["Order Date", "Ship Date"]].dtypes

Order Date    datetime64[us]
Ship Date     datetime64[us]
dtype: object

In [28]:
# Total Sales
total_sales = orders["Sales"].sum()
total_sales

np.float64(12642501.909880001)

In [29]:
# Total Profit
total_profit = orders["Profit"].sum()
total_profit

np.float64(1467457.29128)

In [30]:
# Total Quantity Sold
total_quantity = orders["Quantity"].sum()
total_quantity

np.int64(178312)

In [31]:
# Average Discount
average_discount = orders["Discount"].mean()
average_discount

np.float64(0.14290754533047378)

In [32]:
# Total Shipping Cost
total_shipping_cost = orders["Shipping Cost"].sum()
total_shipping_cost

np.float64(1358085.7034)

In [33]:
# Calculating shipping Duration
orders["Shipping Duration"] = (orders["Ship Date"] - orders["Order Date"]).dt.days

In [34]:
orders["Ship Date"] - orders["Order Date"]

0       7 days
1       3 days
2       7 days
3       7 days
4       3 days
         ...  
51285   5 days
51286   2 days
51287   6 days
51288   5 days
51289   5 days
Length: 51290, dtype: timedelta64[us]

In [35]:
orders[["Order Date", "Ship Date", "Shipping Duration"]].head()

,Order Date,Ship Date,Shipping Duration
0,2017-03-22,2017-03-29,7
1,2015-09-01,2015-09-04,3
2,2017-03-22,2017-03-29,7
3,2017-03-22,2017-03-29,7
4,2015-09-01,2015-09-04,3


In [36]:
# Calculating the profit margin
orders["Profit Margin"] = ( orders["Profit"]/ orders["Sales"]) *100


In [37]:
orders[["Sales", "Profit", "Profit Margin"]].head()

,Sales,Profit,Profit Margin
0,731.82,102.42,13.995245
1,243.54,104.49,42.904656
2,346.32,13.77,3.976091
3,169.68,79.68,46.958982
4,203.88,24.36,11.948205


In [38]:
# Profit margin orders in descending order
orders.nlargest(10, "Profit Margin")[
    ["Order ID", "Product Name", "Sales", "Profit", "Profit Margin"] 
]

,Order ID,Product Name,Sales,Profit,Profit Margin
241,AG-2017-CL25653-42895,"KitchenAid Refrigerator, White",1052.16,526.08,50.0
780,AM-2014-HM49806-41986,"Advantus Door Stop, Durable",42.06,21.03,50.0
1115,IN-2015-MP179657-42350,"Fellowes File Cart, Single Width",274.68,137.34,50.0
1721,IN-2014-RC198257-41897,"Novimex Legal Exhibit Labels, 5000 Label Set",21.72,10.86,50.0
1755,IN-2017-DB136157-43073,"Fellowes File Cart, Single Width",274.68,137.34,50.0
3763,ES-2014-LB167358-41958,"Tenex Trays, Wire Frame",319.68,159.84,50.0
4014,IN-2015-CR1273011-42142,"OIC Push Pins, 12 Pack",100.38,50.19,50.0
4204,BO-2015-AM70513-42208,"Safco Library with Doors, Pine",391.14,195.57,50.0
4273,ES-2017-KC1625514-43056,"OIC Clamps, 12 Pack",96.00,48.00,50.0
4304,ES-2016-JM1526514-42609,"Deflect-O Light Bulb, Durable",64.08,32.04,50.0


In [39]:
orders.nsmallest(10, "Profit Margin")[["Order ID", "Product Name", "Sales", "Profit", "Profit Margin"]]

,Order ID,Product Name,Sales,Profit,Profit Margin
16441,IT-2015-MW1823548-42324,"Chromcraft Coffee Table, Fully Assembled",241.704,-1144.116,-473.354185
39004,IT-2015-NB18580139-42023,"Deflect-O Clock, Black",70.434,-271.236,-385.092427
31529,ID-2015-BG1103597-42147,"Hon Coffee Table, Fully Assembled",63.474,-244.386,-385.017487
31692,ID-2014-GP1474097-41776,"Hon Coffee Table, Fully Assembled",126.948,-488.772,-385.017487
31728,ID-2016-TN2104097-42469,"Lesro Training Table, Adjustable Height",108.396,-390.264,-360.035426
16434,IT-2015-BF1121548-42155,"Bevis Conference Table, Fully Assembled",555.138,-1924.542,-346.678123
35261,ID-2016-BK11260118-42506,"Lesro Coffee Table, Rectangular",99.828,-344.412,-345.005409
31667,ID-2014-BG1174097-41881,"Chromcraft Round Table, Fully Assembled",374.856,-1218.384,-325.027210
44114,CA-2015-MG17695140-42163,Hoover Shoulder Vac Commercial Portable Vacuum,143.128,-393.602,-275.000000
48972,CA-2015-BB11545140-42329,Kensington 6 Outlet SmartSocket Surge Protector,24.588,-67.617,-275.000000


In [40]:
orders_returns = orders.merge(returns,
                              on= "Order ID",
                              how = "left")

In [41]:
orders_returns.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,...,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority,Shipping Duration,Profit Margin,Returned,Region_y
0,24599,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,731.82,2,0.0,102.42,39.66,Medium,7,13.995245,Yes,Southern Asia
1,29465,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,...,243.54,9,0.0,104.49,18.72,Medium,3,42.904656,NaN,NaN
2,24598,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,346.32,3,0.0,13.77,14.10,Medium,7,3.976091,Yes,Southern Asia
3,24597,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,169.68,4,0.0,79.68,11.01,Medium,7,46.958982,Yes,Southern Asia
4,29464,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,...,203.88,4,0.0,24.36,5.72,Medium,3,11.948205,NaN,NaN


In [42]:
orders_complete = orders_returns.merge(people,
                                       left_on= "Region_x",
                                       right_on= "Region",
                                       how="left"
)

In [43]:
orders_complete.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,...,Discount,Profit,Shipping Cost,Order Priority,Shipping Duration,Profit Margin,Returned,Region_y,Person,Region
0,24599,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,0.0,102.42,39.66,Medium,7,13.995245,Yes,Southern Asia,Chandrakant Chaudhri,Southern Asia
1,29465,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,...,0.0,104.49,18.72,Medium,3,42.904656,NaN,NaN,Chandrakant Chaudhri,Southern Asia
2,24598,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,0.0,13.77,14.10,Medium,7,3.976091,Yes,Southern Asia,Chandrakant Chaudhri,Southern Asia
3,24597,IN-2017-CA120551-42816,2017-03-22,2017-03-29,Standard Class,CA-120551,Cathy Armstrong,Home Office,NaN,Herat,...,0.0,79.68,11.01,Medium,7,46.958982,Yes,Southern Asia,Chandrakant Chaudhri,Southern Asia
4,29464,ID-2015-BD116051-42248,2015-09-01,2015-09-04,Second Class,BD-116051,Brian Dahlen,Consumer,NaN,Herat,...,0.0,24.36,5.72,Medium,3,11.948205,NaN,NaN,Chandrakant Chaudhri,Southern Asia


In [44]:
# Sales by Market
sales_by_market = orders.groupby("Market")["Sales"].sum()
sales_by_market

Market
Africa          7.837732e+05
Asia Pacific    4.042658e+06
Europe          3.287336e+06
LATAM           2.164605e+06
USCA            2.364129e+06
Name: Sales, dtype: float64

In [45]:
# Sales and profit by Market 
market_performance = orders.groupby("Market").agg(
    Total_Sales=("Sales", "sum"),
    Total_Profit=("Profit", "sum"),
    Total_Quantity=("Quantity", "sum"),
    Average_Discount=("Discount", "mean")
)

market_performance

,Total_Sales,Total_Profit,Total_Quantity,Average_Discount
Market,,,,
Africa,7.837732e+05,88871.63100,10564,0.156704
Asia Pacific,4.042658e+06,403176.03800,48597,0.180998
Europe,3.287336e+06,449551.72350,41919,0.090890
LATAM,2.164605e+06,221643.48708,38526,0.135531
USCA,2.364129e+06,304214.41170,38706,0.150423


In [46]:
# Performance by Region
region_performance = orders.groupby("Region").agg(
    Total_Sales=("Sales", "sum"),
    Total_Profit=("Profit", "sum"),
    Total_Quantity=("Quantity", "sum")
)

region_performance

,Total_Sales,Total_Profit,Total_Quantity
Region,,,
Canada,6.692817e+04,17817.39000,833
Caribbean,3.242809e+05,34571.32104,6299
Central Africa,1.436300e+05,35383.71000,1553
Central America,1.223101e+06,158981.64816,20882
Central Asia,1.931146e+04,-7282.01100,465
Central US,5.012399e+05,39706.36250,8780
Eastern Africa,1.278560e+05,21900.90900,1688
Eastern Asia,8.550594e+05,167101.85100,8922
Eastern Europe,3.100334e+05,77084.88000,3609


In [47]:
# Performance by Country
country_performance = orders.groupby("Country").agg(
    Total_Sales=("Sales", "sum"),
    Total_Profit=("Profit", "sum"),
    Total_Quantity=("Quantity", "sum")
)

country_performance

,Total_Sales,Total_Profit,Total_Quantity
Country,,,
Afghanistan,21673.32000,5460.30000,228
Albania,3888.12000,709.32000,40
Algeria,36091.59000,9106.50000,454
Angola,25554.00000,6494.97000,317
Argentina,57511.78328,-18693.79672,1504
...,...,...,...
Vietnam,65210.06580,-1697.31420,991
Western Sahara,666.36000,300.42000,4
Yemen,2465.71200,-3706.45800,71


In [49]:
# Performance by Category

category_performance = orders.groupby("Category").agg(
    Total_Sales=("Sales", "sum"),
    Total_Profit=("Profit", "sum"),
    Total_Quantity=("Quantity", "sum")
)

category_performance

,Total_Sales,Total_Profit,Total_Quantity
Category,,,
Furniture,4.110452e+06,285082.73020,34892
Office Supplies,3.787493e+06,518595.82790,108244
Technology,4.744557e+06,663778.73318,35176


In [52]:
# Performance by Sub-Category
subcategory_performance = orders.groupby("Sub-Category").agg(
    Total_Sales=("Sales", "sum"),
    Total_Profit=("Profit", "sum"),
    Total_Quantity=("Quantity", "sum")
)

subcategory_performance


,Total_Sales,Total_Profit,Total_Quantity
Sub-Category,,,
Accessories,7.492370e+05,129626.30620,10946
Appliances,1.010536e+06,141562.58770,6026
Art,3.716132e+05,57829.85930,16215
Binders,4.618694e+05,72433.15160,21403
Bookcases,1.466572e+06,161924.41950,8310
Chairs,1.501682e+06,140396.26750,12336
Copiers,1.509436e+06,258567.54818,7454
Envelopes,1.692175e+05,28849.48730,8210
Fasteners,8.949505e+04,13844.28890,9051


In [53]:
# Performance by segment
segment_performance = orders.groupby("Segment").agg(
    Total_Sales=("Sales", "sum"),
    Total_Profit=("Profit", "sum"),
    Total_Quantity=("Quantity", "sum")
)

segment_performance

,Total_Sales,Total_Profit,Total_Quantity
Segment,,,
Consumer,6.507949e+06,749239.78206,92157
Corporate,3.824698e+06,441208.32866,53565
Home Office,2.309855e+06,277009.18056,32590


In [54]:
#Category with highest sales
category_performance.sort_values(
    "Total_Sales",
    ascending=False
)

,Total_Sales,Total_Profit,Total_Quantity
Category,,,
Technology,4.744557e+06,663778.73318,35176
Furniture,4.110452e+06,285082.73020,34892
Office Supplies,3.787493e+06,518595.82790,108244


In [55]:
# Category with highest profit
category_performance.sort_values(
    "Total_Profit",
    ascending=False
)

,Total_Sales,Total_Profit,Total_Quantity
Category,,,
Technology,4.744557e+06,663778.73318,35176
Office Supplies,3.787493e+06,518595.82790,108244
Furniture,4.110452e+06,285082.73020,34892


In [57]:
# Pivot table for sales by Market and Category
market_category_pivot = pd.pivot_table(
    orders,
    values="Sales",
    index="Market",
    columns="Category",
    aggfunc="sum"
)

market_category_pivot

Category,Furniture,Office Supplies,Technology
Market,,,
Africa,1.946506e+05,2.667555e+05,3.223670e+05
Asia Pacific,1.461551e+06,1.043652e+06,1.537455e+06
Europe,8.901066e+05,1.163661e+06,1.233569e+06
LATAM,8.119706e+05,5.639207e+05,7.887138e+05
USCA,7.521728e+05,7.495034e+05,8.624528e+05


In [58]:
# Products containing "chair"
chair_products = orders[
    orders["Product Name"].str.contains(
        "Chair",
        case=False,
        na=False
    )
]

In [59]:
chair_products["Product Name"].nunique()

206

In [61]:
# Customers containing "Smith" in their name
smith_customers = orders[
    orders["Customer Name"].str.contains(
        "Smith",
        case=False,
        na=False
    )
]


In [62]:
smith_customers["Customer Name"].nunique()

5

In [63]:
# First word in the product name
orders["First Word"] = orders["Product Name"].str.split().str[0]

In [64]:
orders[["Product Name", "First Word"]].head()

,Product Name,First Word
0,"Ikea Library with Doors, Mobile",Ikea
1,"Acme Scissors, Easy Grip",Acme
2,"Epson Receipt Printer, White",Epson
3,"Rubbermaid Door Stop, Erganomic",Rubbermaid
4,"Cameo Interoffice Envelope, with clear poly wi...",Cameo


In [65]:
# Five most common five words
orders["First Word"].value_counts().head(5)

First Word
Avery    1916
Eldon    1471
Hon      1330
Tenex    1323
Smead    1303
Name: count, dtype: int64

In [66]:
# Extracting the year
orders["Year"] = orders["Order Date"].dt.year

In [67]:
orders[["Order Date", "Year"]].head()

,Order Date,Year
0,2017-03-22,2017
1,2015-09-01,2015
2,2017-03-22,2017
3,2017-03-22,2017
4,2015-09-01,2015


In [69]:
#Extracting the month
orders["Month"] = orders["Order Date"].dt.month


In [71]:
orders[["Order Date", "Month"]].head()

,Order Date,Month
0,2017-03-22,3
1,2015-09-01,9
2,2017-03-22,3
3,2017-03-22,3
4,2015-09-01,9


In [73]:
#Extracting the month name
orders["Month Name"] = orders["Order Date"].dt.month_name()

In [74]:
# Sales by year
sales_by_year = orders.groupby("Year")["Sales"].sum()
sales_by_year

Year
2014    2.259451e+06
2015    2.677439e+06
2016    3.405746e+06
2017    4.299866e+06
Name: Sales, dtype: float64

In [75]:
# Profit by year
profit_by_year = orders.groupby("Year")["Profit"].sum()
profit_by_year

Year
2014    248940.81154
2015    307415.27910
2016    406935.23018
2017    504165.97046
Name: Profit, dtype: float64

In [76]:
# Sales by month
sales_by_month = orders.groupby("Month")["Sales"].sum()
sales_by_month

Month
1     6.767889e+05
2     5.546499e+05
3     7.605180e+05
4     6.959784e+05
5     9.154928e+05
6     1.262081e+06
7     7.523063e+05
8     1.294315e+06
9     1.431753e+06
10    1.185228e+06
11    1.546728e+06
12    1.566662e+06
Name: Sales, dtype: float64

In [77]:
# Monthly sales across years
monthly_sales = orders.groupby(
    ["Year", "Month"]
)["Sales"].sum()

monthly_sales

Year  Month
2014  1         98898.48886
      2        103717.92328
      3        135746.40206
      4        114332.96318
      5        158228.32880
      6        207571.54352
      7        118434.88462
      8        208063.28372
      9        284587.74846
      10       216114.56748
      11       293947.35216
      12       319807.40940
2015  1        137435.97514
      2         98854.96208
      3        163076.77116
      4        161052.26952
      5        208364.89124
      6        256175.69842
      7        145236.78512
      8        303142.94238
      9        289389.16564
      10       252939.85020
      11       323512.41690
      12       338256.96660
2016  1        199185.90738
      2        167239.65040
      3        198594.03012
      4        177821.31684
      5        260498.56470
      6        396519.61190
      7        229928.95200
      8        326488.78936
      9        376619.24568
      10       293406.64288
      11       373989.36010
      12

In [78]:
# Monthly profit
monthly_profit = orders.groupby(
    ["Year", "Month"]
)["Profit"].sum()

monthly_profit

Year  Month
2014  1         8321.80096
      2        14323.06088
      3        13089.62696
      4        13211.11178
      5        14644.08230
      6        20940.59862
      7         4040.60682
      8        26452.99742
      9        35044.52286
      10       27597.95588
      11       33242.88306
      12       38031.56400
2015  1        10306.71254
      2        15095.02128
      3        17992.91756
      4        17366.96722
      5        29876.70374
      6        34407.15362
      7        15585.38842
      8        43573.87858
      9        27776.18034
      10       30662.88270
      11       31820.72180
      12       32950.75130
2016  1        26810.55968
      2        23762.49610
      3        23433.77462
      4        19462.03844
      5        28495.69410
      6        45478.41340
      7        28863.82720
      8        31023.66846
      9        38905.66778
      10       42433.22258
      11       48062.99670
      12       50202.87112
2017  1        2

In [79]:
# Highest Sales month
monthly_sales.sort_values(ascending=False).head(10)

Year  Month
2017  11       555279.02700
      12       503143.69348
      9        481157.24370
      8        456619.94236
      10       422766.62916
2016  12       405454.37802
2017  6        401814.06310
2016  6        396519.61190
      9        376619.24568
      11       373989.36010
Name: Sales, dtype: float64

In [80]:
#Average Shipping duration
orders["Shipping Duration"].mean()

np.float64(3.970403587443946)

In [81]:
# Shipping duration bu ship mode
shipping_by_mode = orders.groupby("Ship Mode").agg(
    Average_Shipping_Days=("Shipping Duration", "mean"),
    Total_Orders=("Order ID", "nunique")
)

shipping_by_mode

,Average_Shipping_Days,Total_Orders
Ship Mode,,
First Class,2.182678,3845
Same Day,0.037394,1349
Second Class,3.230478,5146
Standard Class,4.999415,15405


In [ ]:
# Hierarchical indexing
region_category = orders.groupby(
    ["Region", "Category"]
)["Sales"].sum()

In [83]:
region_category

Region          Category       
Canada          Furniture           10595.2800
                Office Supplies     30034.0800
                Technology          26298.8100
Caribbean       Furniture          118372.4000
                Office Supplies     89575.4160
                                      ...     
Western Europe  Office Supplies    601443.6450
                Technology         656637.1440
Western US      Furniture          252434.4675
                Office Supplies    221031.5250
                Technology         251991.8320
Name: Sales, Length: 69, dtype: float64

In [84]:
region_category.index

MultiIndex([(           'Canada',       'Furniture'),
            (           'Canada', 'Office Supplies'),
            (           'Canada',      'Technology'),
            (        'Caribbean',       'Furniture'),
            (        'Caribbean', 'Office Supplies'),
            (        'Caribbean',      'Technology'),
            (   'Central Africa',       'Furniture'),
            (   'Central Africa', 'Office Supplies'),
            (   'Central Africa',      'Technology'),
            (  'Central America',       'Furniture'),
            (  'Central America', 'Office Supplies'),
            (  'Central America',      'Technology'),
            (     'Central Asia',       'Furniture'),
            (     'Central Asia', 'Office Supplies'),
            (     'Central Asia',      'Technology'),
            (       'Central US',       'Furniture'),
            (       'Central US', 'Office Supplies'),
            (       'Central US',      'Technology'),
            (   'Eastern Afr

In [89]:
# Sales for each category in Africa
africa_category_sales = region_category.loc["Central US"]
africa_category_sales

Category
Furniture          163692.5758
Office Supplies    167131.0030
Technology         170416.3120
Name: Sales, dtype: float64

In [88]:
# Sales for furniture in the Central US
furniture_sales_central_us = region_category.loc[("Central US", "Furniture")]
furniture_sales_central_us

np.float64(163692.5758)

**Overall performance**

In [90]:
print("Total Sales:", total_sales)
print("Total Profit:", total_profit)
print("Total Quantity:", total_quantity)
print("Average Discount:", average_discount)
print("Total Shipping Cost:", total_shipping_cost)

Total Sales: 12642501.909880001
Total Profit: 1467457.29128
Total Quantity: 178312
Average Discount: 0.14290754533047378
Total Shipping Cost: 1358085.7034


**Best Market**

In [91]:
market_performance.sort_values(
    "Total_Profit",
    ascending=False
)

,Total_Sales,Total_Profit,Total_Quantity,Average_Discount
Market,,,,
Europe,3.287336e+06,449551.72350,41919,0.090890
Asia Pacific,4.042658e+06,403176.03800,48597,0.180998
USCA,2.364129e+06,304214.41170,38706,0.150423
LATAM,2.164605e+06,221643.48708,38526,0.135531
Africa,7.837732e+05,88871.63100,10564,0.156704


**Worst Market**

In [92]:
market_performance.sort_values(
    "Total_Profit",
    ascending=True
)

,Total_Sales,Total_Profit,Total_Quantity,Average_Discount
Market,,,,
Africa,7.837732e+05,88871.63100,10564,0.156704
LATAM,2.164605e+06,221643.48708,38526,0.135531
USCA,2.364129e+06,304214.41170,38706,0.150423
Asia Pacific,4.042658e+06,403176.03800,48597,0.180998
Europe,3.287336e+06,449551.72350,41919,0.090890


**Best Category**

In [93]:
category_performance.sort_values(
    "Total_Profit",
    ascending=False
)

,Total_Sales,Total_Profit,Total_Quantity
Category,,,
Technology,4.744557e+06,663778.73318,35176
Office Supplies,3.787493e+06,518595.82790,108244
Furniture,4.110452e+06,285082.73020,34892


**Worst Category**

In [94]:
category_performance.sort_values(
    "Total_Profit",
    ascending=True
)

,Total_Sales,Total_Profit,Total_Quantity
Category,,,
Furniture,4.110452e+06,285082.73020,34892
Office Supplies,3.787493e+06,518595.82790,108244
Technology,4.744557e+06,663778.73318,35176


**Best Region**

In [95]:
region_performance.sort_values(
    "Total_Profit",
    ascending=False
)

,Total_Sales,Total_Profit,Total_Quantity
Region,,,
Western Europe,1.731930e+06,218433.50850,22262
Eastern Asia,8.550594e+05,167101.85100,8922
Southern Asia,8.665727e+05,159336.42700,9109
Central America,1.223101e+06,158981.64816,20882
Oceania,1.100185e+06,120089.11200,12838
Western US,7.254578e+05,108418.44890,12266
Eastern US,6.787812e+05,91522.78000,10618
Northern Europe,6.367792e+05,83923.91700,8177
Eastern Europe,3.100334e+05,77084.88000,3609


## Final Business Summary

The Global Superstore analysis shows that the business generated total sales of approximately **12.64 million** and total profit of approximately **1.47 million**, with **178,312 units** sold. The overall average discount was **14.29%**, while total shipping costs amounted to approximately **1.36 million**.

### Key Business Insights

**1. Market Performance**  
Europe was the most profitable market, generating approximately **449,552 in profit**, despite having the lowest average discount of **9.09%**. Asia Pacific generated the highest sales at approximately **4.04 million**, but its profit was lower than Europe's at approximately **403,176**. Asia Pacific also had the highest average discount at **18.10%**. This suggests that higher sales do not necessarily result in higher profits and that excessive discounting may reduce profitability.

**2. Category Performance**  
Technology was the most profitable category, generating approximately **663,779 in profit** from **4.74 million in sales**. Office Supplies had the highest sales volume, with **108,244 units sold**, but generated a lower profit of approximately **518,596**. Furniture generated approximately **4.11 million in sales** but had the lowest profit at approximately **285,083**. This indicates that sales volume alone does not determine profitability.

**3. Overall Business Performance**  
The analysis indicates that the business performs strongly in terms of sales and profit, but there are differences in profitability across markets and product categories. Markets and categories with higher sales should not automatically be considered the most successful because discounts, costs, and profit margins also affect overall performance.

### Recommendations

- **Review discounting strategies**, particularly in markets such as Asia Pacific where the average discount is relatively high.
- **Prioritize Technology products**, as they generate the highest total profit.
- **Investigate Furniture profitability** to identify whether high costs, discounts, or low margins are reducing profits.
- **Focus on profitability rather than sales volume alone** when evaluating markets and product categories.
- **Monitor shipping costs** and shipping duration to identify opportunities to reduce operational costs and improve efficiency.